circuits/cpp/src/aztec3/circuits/rollup/root/native_root_rollup_circuit.cpp

```diff
@@ -63,6 +112,31 @@ RootRollupPublicInputs root_rollup_circuit(DummyComposer& composer, RootRollupIn
                                                     0,
                                                     "historic contract tree roots insertion");
 
+    // Check correct l1 to l2 tree given
+    // Compute subtree inserting l1 to l2 messages
+    auto l1_to_l2_subtree_root = calculate_subtree(rootRollupInputs.l1_to_l2_messages);
+
+    // // Insert subtree into the l1 to l2 data tree
+    const auto empty_l1_to_l2_subtree_root = components::calculate_empty_tree_root(L1_TO_L2_MSG_SUBTREE_DEPTH);
+    auto new_l1_to_l2_messages_tree_root =
```

Comment:
Possibly rename this to snapshot for clarity. Looks strange when we later have `root.root`